crates/bevy_ptr/src/lib.rs

```diff
@@ -357,3 +374,36 @@ impl<'a, T> UnsafeCellDeref<'a, T> for &'a UnsafeCell<T> {
         self.get().read()
     }
 }
+
+trait EnsureAligned {
+    fn ensure_aligned(self) -> Self;
+}
+
+// Disable this for miri runs as it already checks if pointer to reference
+// casts are properly aligned.
+#[cfg(all(debug_assertions, not(miri)))]
+impl<T: Sized> EnsureAligned for *mut T {
+    #[inline(always)]
```

Comment:
```suggestion
```

It might be better not to inline this function, since it contains a panic. Feel free to ignore.